src/basic/path-util.c

```diff
@@ -448,28 +448,25 @@ char* path_startswith(const char *path, const char *prefix) {
                 return NULL;
 
         for (;;) {
-                size_t a, b;
-
-                path += strspn(path, "/");
-                prefix += strspn(prefix, "/");
+                const char *p, *q;
+                int r, s;
 
-                if (*prefix == 0)
-                        return (char*) path;
+                r = path_find_first_component(&path, accept_dot_dot, &p);
+                if (r < 0)
+                        return NULL;
 
-                if (*path == 0)
+                s = path_find_first_component(&prefix, accept_dot_dot, &q);
```

Comment:
doesn't matter, but in 95% of all cases local variables called `s` refers to a string in functions that primarily deal with strings... maybe rename `k`? 

but really doesn't matter